### PR TITLE
Add confirmed transaction lookups and always track roots

### LINKED ISSUE #3442: getRecentBlockhash JSON RPC API returns different values from different nodes in testnet-beta
STR:
1. Figure the IP addresses of the individual testnet nodes from the latest "testnet-beta" run at https://buildkite.com/solana-labs/testnet-management/builds?branch=v0.12
2. Run the following for a couple nodes, setting `rpc` correctly:
```bash
$ rpc=52.53.220.212:8899
$ while sleep 1; do curl -X POST -H 'Content-Type: application/json' -d '{"jsonrpc":"2.0","id":1, "method":"getRecentBlockhash"}' $rpc; done
```
3. Observe non-overlapping `getRecentBlockhash` values.

It appears that the validator nodes are generating different blockhash values, which means any transaction create via their RPC API will not be accepted by the leader.

### PR DESCRIPTION
**Pull request by sagar-solana**
#### Problem
Nodes use bleeding edge banks for all RPC APIs. This results in some apis reporting different results on different nodes
#### Summary of Changes
Ensure the root is always updated (even when no voting keypair is provided) and only provide confirmed blockhashes to rpc clients.
Fixes #3442

### CODE REVIEW

**Comment by sagar-solana** [comment]
@mvines, I tried this with 3 nodes running off of my PC and it seems to work. Can you help verify blockstreamer ?

File: runtime/src/bank.rs [hunk 1]
```diff
@@ -382,6 +382,13 @@ impl Bank {
     pub fn last_blockhash(&self) -> Hash {
         self.blockhash_queue.read().unwrap().last_hash()
     }
+    pub fn get_confirmed_blockhash(&self) -> Hash {
```
Comment (mvines) [inline runtime/src/bank.rs]: nit: maybe call this `confirmed_last_blockhash()` for consistency with `transaction_count()`/`confirmed_transaction_count()`

File: runtime/src/bank.rs [hunk 2]
```diff
@@ -867,6 +874,13 @@ impl Bank {
     pub fn transaction_count(&self) -> u64 {
         self.transaction_count.load(Ordering::Relaxed) as u64
     }
+    pub fn confirmed_transaction_count(&self) -> u64 {
+        if let Some(bank) = self.parents().last() {
```
Comment (mvines) [inline runtime/src/bank.rs]: Would this one-liner work instead?
```rust
self.parents().last().unwrap_or(self).transaction_count()
````

Comment (sagar-solana) [reply runtime/src/bank.rs]: Not exactly, we've got &self and each parent is &Arc<Bank>

**Pull request merged**